### PR TITLE
Enable Redirect Settings

### PR DESCRIPTION
**Pull request by F4bsi**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
I added the possibility to change the destination of the redirection if the user is not logged in for the main page.
Currently you wil lbe redirected to the login page (if the index page is protected) but I wanted to have the possibility to redirect them to the gallery only page. Maybe this is helpful for other people too.

Wasn't quite sure which type the setting should have. Guess it could also be a select setting. With the input you could redirect the user to almost anywhere, but the select would be easier for the admin...

Hope I did everything right with the new structure and stuff :)

### CODE REVIEW

File: lib/configsetup.inc.php [hunk 1]
```diff
@@ -1629,6 +1629,13 @@
             'name' => 'protect[localhost_index]',
             'value' => $config['protect']['localhost_index'],
         ],
+        'index_redirect' => [
```
Comment (andi34) [inline lib/configsetup.inc.php]: protect_index_redirect

File: lib/configsetup.inc.php [hunk 2]
```diff
@@ -1629,6 +1629,13 @@
...
+            'view' => 'advanced',
+            'type' => 'input',
+	        'placeholder' => $defaultConfig['protect']['index_redirect'],
```
Comment (andi34) [inline lib/configsetup.inc.php]: Space

File: resources/lang/en.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
   "authentication:login_username": "Username",
   "authentication:protect_admin": "Protect admin panel",
   "authentication:protect_index": "Protect start screen",
+  "authentication:index_redirect": "Destination from start screen when not logged in",
```
Comment (andi34) [inline resources/lang/en.json]: :protect_

**Review by andi34 — CHANGES_REQUESTED** [review]
Great idea! There's some nit-picks, see the comments.
Redirect should also follow the config inside the `livechroma.php`:
https://github.com/andi34/photobooth/blob/9cbc41a5ea70d2f5126c58d9d12ce60dfa3ecdb0/livechroma.php#L29


Maybe let's make it choosable instead an input?
```
        'protect_index_redirect' => [
            'view' => 'advanced',
            'type' => 'select',
            'name' => 'protect[index_redirect]',
            'placeholder' => $defaultConfig['protect']['index_redirect'],
            'options' => [
                'login' => 'Login',
                'gallery.php' => 'Standalone Gallery',
                'slideshow' => 'Standalone Slideshow',
            ],
            'value' => $config['protect']['index_redirect'],
        ],
```
Or we need to have the possibility to redirect to a completely different page (e.g. `private/my_private.php` or a page somewhere on the internet)?

File: lib/configsetup.inc.php (hunk 2 above)
Comment (F4bsi) [reply lib/configsetup.inc.php]: Hmm I don't see that in my ide. Maybe because I used tabs?

Comment (F4bsi) [reply lib/configsetup.inc.php]:
```suggestion
	      'placeholder' => $defaultConfig['protect']['index_redirect'],
```

**Comment by F4bsi** [comment]
Glad that you like my idea. Crazy how far this project has come over the last 2 years!

I'm not 100% sure about the best way for selecting the redirection. But I guess there would be far less errors if we do a choosable/select option and I would say that for 99% of the users that would be enough. 

Maybe  we could do that for advanced and a free input for experts? But then you would have two inputs for the same thing...

File: lib/configsetup.inc.php [hunk 4]
```diff
@@ -1629,6 +1629,18 @@
...
         ],
+        'protect_index_redirect' => [
+            'view' => 'advanced',
+            'type' => 'select',
+            'options' => [
+                'login' => 'Login',
+                'gallery.php' => 'Standalone Gallery',
+                'slideshow' => 'Standalone Slideshow',
+            ],
+            'placeholder' => $defaultConfig['protect']['index_redirect'],
```
Comment (F4bsi) [inline lib/configsetup.inc.php:1641]: @andi34 do you still need the placeholder here, or is the first option always the default?

**Comment by andi34** [comment]
Maybe as last option to choose set to `Private` which will redirect to `private/index.php` from there you'll be able to redirect where ever you want to?

File: lib/configsetup.inc.php (hunk 4 above)
Comment (andi34) [reply lib/configsetup.inc.php:1641]: Not sure to be true, used the language section as base and adjusted it

**Comment by andi34** [comment]
Sidenote: if you run `yarn format` before commiting, the formatting will be adjusted if needed :)

File: lib/configsetup.inc.php [hunk 5]
```diff
@@ -1629,6 +1629,18 @@
             'name' => 'protect[localhost_index]',
             'value' => $config['protect']['localhost_index'],
         ],
+        'protect_index_redirect' => [
+            'view' => 'advanced',
+            'type' => 'select',
+            'options' => [
```
Comment (andi34) [inline lib/configsetup.inc.php:1635]:
```
            'options' => [
                'login' => 'Login',
                'gallery.php' => 'Standalone Gallery',
                'slideshow' => 'Standalone Slideshow',
                'private' => 'Private (private/index.php)',
            ],
```

**Review by andi34 — CHANGES_REQUESTED** [review]
Should be good to go if private is added to options

**Comment by F4bsi** [comment]
Alright :) 
Already added it in my local repo. Will push it tonight when I'm back home.

**Comment by F4bsi** [comment]
Hey @andi34, 

I added the private option (had a little bit different wording, is that ok?)
Also not quite sure about the help text and title, but I guess that it would be easy to change.

**Review by andi34 — APPROVED** [review]

**Comment by andi34** [comment]
Thanks! And yeah, text can be adjusted easily.

**Pull request merged**